### PR TITLE
add function container

### PR DESCRIPTION
**Pull request by jan-janssen**
This was a demonstration, primarily for the IPAM workshop on workflows, how we could combine the concepts of ScriptJob and GenericMaster. It still requires the integration with `cloudpickle` and as we discussed last time it most likely makes more sense to move these developments first to `pyiron_contrib`.

### CODE REVIEW

File: pyiron_base/jobs/functioncontainer.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from pyiron_base.jobs.job.generic import GenericJob
+
+
+class FunctionContainer(GenericJob):
```
Comment (liamhuber) [inline pyiron_base/jobs/functioncontainer.py:4]: Do you mean `PythonTemplateJob`?

File: pyiron_base/jobs/functioncontainer.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self._python_only_job = True
+
+    def run_static(self):
+        for kwargs in self.input:
+            self.function(**kwargs)
```
Comment (liamhuber) [inline pyiron_base/jobs/functioncontainer.py:11]: `GenericJob` doesn't have an `input` class, so it's not totally clear what `in` is doing here, but naively I would expect this to call `self.function` multiple times. Do you mean something like this?

```suggestion
        self.function(**self.input.to_dict())
```

File: pyiron_base/jobs/functioncontainer.py (hunk 1 above)
Comment (jan-janssen) [reply pyiron_base/jobs/functioncontainer.py:4]: The issue is that `DataContainer` cannot store projects, that is why I cannot use the `PythonTemplateJob`.

Comment (pmrv) [reply pyiron_base/jobs/functioncontainer.py:4]: What is missing to store projects in `DataContainer` and how does this sidestep the problem?  I don't think `Project` comes with a `to_hdf`, but that also seems easy to add (just save the path).

**Pull request closed without merging**